graphql/schema.graphql

```diff
@@ -41,14 +41,17 @@ type Patch {
   author: String!
   version: String!
   status: String!
-  createTime: Time!
-  startTime: Time
-  finishTime: Time
   variants: [String!]!
   tasks: [String!]!
   variantsTasks: [VariantTask]!
   activated: Boolean!
   alias: String!
+  duration: PatchDuration
+}
+
+type PatchDuration {
+  makespan: String
```

Comment:
Why not return these as durations and let the formatting function Arjun added in spruce format them?

Reply:
because graphql gives the power to deliver data in a readily-consumable format. why do logic on the client when you can deliver the data in the shape the client needs.

Reply:
I would argue the inverse of your question - why add another library to spruce when you can just use the built in powers of Go?

Reply:
i think the formatting function arjun added in spruce shoudl be deleted, remove the 100kb of code that lib adds to achieve one function, and do all formating in resolvers.

Reply:
follow up: the function arjun wrote only works for ms

Reply:
I think @john-m-liu is implying that makespan can return a Duration is in the form of milliseconds

Reply:
yeah def confused myself there -- i'm doing a PR at the same time that works with dates that woudl require moment. 

i'd like to deliver as much data in a readily-consumable form as possible going forward. query the data from gql and it's ready to rendered in the UI. that means doing stuff like this in the resolvers.

Reply:
I can't believe that an integer-to-ascii function is going to bloat the app. 

Are we going to use moment.js at all? Do we expect to?

 Do we have a policy/strategy for considering weather 3rd party libraries are worth it? How do we evaluate that cost? Without some kind of strategy there, it seems like we're setting ourselves up for premature optimization problems, and spending developer time when we could just acquire dependencies (which I think is the wrong trade off most of the time.)

Reply:
moment.js is a massive library, usually the largest of an app's dependencies, and i'd like to avoid using it if possible. formatting the dates in this way on the front end would require moment. that is a very small use case to import a very large library. 

i love using libraries but formatting these dates on the server is easy and saves a material dependency being added to bundle size.

Reply:
ugh, i keep mixing up my prs! moment is not required for this formatting. i don't see why we can't format fields on the server and send them to the client ready to be consumed.